dbt_af/conf/config.py

```diff
@@ -5,6 +5,30 @@
 import pendulum
 
 
+@attrs.define(frozen=True)
+class AfCallbacksConfig:
+    """
+    Config to define callbacks functions for airflow DAGs and tasks
+
+    :param task_on_success_callback: success callback function for task (invoked when the task succeeds)
+    :param task_on_failure_callback: failure callback function for task (invoked when the airflow task fails)
+    :param task_on_retry_callback: retry callback function for task (invoked when the task is up for retry)
+    :param task_on_execute_callback: execute callback function for task (invoked right before task begins executing)
+    :param dag_on_failure_callback: failure callback function for DAG (invoked when the airflow task fails)
+    :param dag_on_success_callback: success callback function for DAG (invoked when the task succeeds)
+    :param dag_sla_miss_callback: sla miss callback function for DAG (invoked when a task misses its defined SLA)
+    """
+
+    task_on_success_callback: callable = attrs.field(default=None)
+    task_on_failure_callback: callable = attrs.field(default=None)
+    task_on_retry_callback: callable = attrs.field(default=None)
+    task_on_execute_callback: callable = attrs.field(default=None)
+
+    dag_on_failure_callback: callable = attrs.field(default=None)
+    dag_on_success_callback: callable = attrs.field(default=None)
+    dag_sla_miss_callback: callable = attrs.field(default=None)
```

Comment:
Multiple function registrations have been added for a single event. The tuple data type has been used because lists are not hashable, and the attrs.define(frozen=True) check cannot be performed. Later, in the `common/af_callbacks/merge_dicts_by_key` method, we convert these tuples into lists for further processing.